c/polyjuice.h

```diff
@@ -1010,6 +1019,100 @@ int handle_transfer(gw_context_t* ctx,
   return 0;
 }
 
+//read eoa type hash from config
+int load_eoa_type_hash(gw_context_t* ctx, uint8_t eoa_type_hash[GW_KEY_BYTES]) {
+  mol_seg_t rollup_config_seg;
+  rollup_config_seg.ptr = ctx->rollup_config;
+  rollup_config_seg.size = ctx->rollup_config_size;
+
+  mol_seg_t allowed_eoa_list_seg =
+      MolReader_RollupConfig_get_allowed_eoa_type_hashes(&rollup_config_seg);
+  mol_seg_res_t allowed_type_hash_res =
+      MolReader_AllowedTypeHashVec_get(&allowed_eoa_list_seg, 0);
+
+  mol_seg_t code_hash_seg =
+      MolReader_AllowedTypeHash_get_hash(&allowed_type_hash_res.seg);
+  memcpy(eoa_type_hash, code_hash_seg.ptr, 32);
+  return 0;
+}
+
+int handle_native_token_transfer(gw_context_t* ctx, uint32_t from_id, uint256_t value,
+                        uint64_t gas_used) {
+  if (g_creator_account_id == UINT32_MAX) {
+    ckb_debug("[handle_native_token_transfer] g_creator_account_id wasn't set.");
+    return ERROR_NATIVE_TOKEN_TRANSFER;
+  }
+  if (!g_eoa_transfer_flag) {
+    ckb_debug("[handle_native_token_transfer] g_eoa_transfer_flag wasn't set.");
+    return ERROR_NATIVE_TOKEN_TRANSFER;
+  }
+  evmc_address zero_address = {0};
+  if (memcmp(zero_address.bytes, g_eoa_transfer_to_address.bytes, 20) == 0) {
+    ckb_debug("[handle_native_token_transfer] g_eoa_transfer_to_address wasn't set.");
+    return ERROR_NATIVE_TOKEN_TRANSFER;
+  }
+
+  int ret = 0;
+  //check to_address is created
+  gw_reg_addr_t to_addr = new_reg_addr(g_eoa_transfer_to_address.bytes);
+  uint8_t script_hash[GW_KEY_BYTES] = {0};
+  ret = ctx->sys_get_script_hash_by_registry_address(ctx, &to_addr,
+          script_hash);
+  if (ret == GW_ERROR_NOT_FOUND) {
+    //to_address not exists, create new EOA account first
```

Comment:
The `sudt_transfer` function supports transferring to an unactivated eth_address.
Is that necessary to create new EOA account first?

Reply:
I didn't know that.
I found this in _sudt_transfer:
```c
  /* check to account */
  uint256_t to_balance = {0};
  ret = _sudt_get_balance(ctx, sudt_id, to_addr, &to_balance);
  if (ret != 0) {
    printf("transfer: can't get receiver's balance");
    return ret;
  }
```
Are you sure sudt_transfer supports transferring to an unactivated eth address?

Reply:
> Are you sure sudt_transfer supports transferring to an unactivated eth address?

yes.
`to_addr` is `gw_reg_addr_t`, it didn't require a Godwoken account ID.

---
But, maybe activating the receiver at this time is a good idea?
We could discuss.